test/test_postman.py

```diff
@@ -48,26 +65,30 @@ def test_init_ganache():
 
 @pytest.mark.web3_deploy
 def test_deploy_l1_contracts():
-    """Deploys Ethereum contracts in the Ganache instance, including the L1L2Example and MockStarknetMessaging contracts"""
+    """Deploys Ethereum contracts in the Hardhat testnet instance, including the L1L2Example and MockStarknetMessaging contracts"""
 
-    global MESSAGING_CONTRACT_ADDRESS # pylint: disable=global-statement
-    global L1L2_EXAMPLE_CONTRACT_ADDRESS # pylint: disable=global-statement
+    global MESSAGING_CONTRACT_DEFINITION # pylint: disable=global-statement
+    global L1L2_EXAMPLE_CONTRACT_DEFINITION # pylint: disable=global-statement
+    global WEB3 # pylint: disable=global-statement disable=global-variable-not-assigned
 
-    args = "cd test && truffle migrate && cd .."
-    subprocess.run(args, shell=True, encoding="utf-8", check=False, capture_output=True)
-    messaging_contract = json.loads(load_file_content(STARKNET_MESSAGING_PATH))
-    l1l2_example_contract = json.loads(load_file_content(L1L2_EXAMPLE_ETH_PATH))
+    install_solc(Version('0.6.12'))
+    set_solc_version(Version('0.6.12'))
+    install_solc_pragma("pragma solidity ^0.6.12;")
+    messaging_contract = load_file_content(STARKNET_MESSAGING_PATH)
+    l1l2_example_contract = load_file_content(L1L2_EXAMPLE_ETH_PATH)
 
-    MESSAGING_CONTRACT_ADDRESS = messaging_contract["networks"]["32"]["address"]
-    L1L2_EXAMPLE_CONTRACT_ADDRESS = l1l2_example_contract["networks"]["32"]["address"]
+    messaging_contract_interface = web3_compile(messaging_contract, os.path.abspath(SOLIDITY_CONTRACTS_PATH))
+    l1l2_example_contract_interface = web3_compile(l1l2_example_contract, os.path.abspath(SOLIDITY_CONTRACTS_PATH))
+    MESSAGING_CONTRACT_DEFINITION = web3_deploy(WEB3,messaging_contract_interface)
```

Comment:
Why do these two variables have the _DEFINITION suffix? Are they contracts or definitions? I find it confusing that a definition holds an address.

Reply:
Renamed